core/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/status/TransportSnapshotsStatusAction.java

```diff
@@ -214,7 +214,14 @@ private SnapshotsStatusResponse buildResponse(SnapshotsStatusRequest request, Li
                 SnapshotId snapshotId = matchedSnapshotIds.get(snapshotName);
                 if (snapshotId == null) {
                     // neither in the current snapshot entries nor found in the repository
-                    throw new SnapshotMissingException(repositoryName, snapshotName);
+                    if (request.ignoreUnavailable()) {
+                        // ignoring unavailable snapshots, so skip over
+                        logger.trace("snapshot status request ignoring snapshot [{}], not found in repository [{}]",
```

Comment:
Its up to you! I'm ok with whatever you think is right, I just figured I'd mention how I'd do it.